### PR TITLE
Define 'DeploymentResponse' to parse Zip Deploy status response

### PR DESCRIPTION
**Pull request by anvillan**
Define new `DeploymentResponse` type to parse Json response from ZipDeploy publish (deployment) operation status.

In this PR:
- Define new `DeploymentResponse` type
- Use the new type when querying the deployment operation status.
- For a failed status, add the `log_url` to the error message
- Add UTs

### CODE REVIEW

File: src/WebSdk/Publish/Tasks/Tasks/ZipDeploy/ZipDeploy.cs [hunk 1]
```diff
@@ -114,16 +114,17 @@ public async System.Threading.Tasks.Task<bool> ZipDeployAsync(string zipToPublis
...
+                    if (deploymentResponse?.Status == DeployStatus.Success)
                     {
                         Log.LogMessage(MessageImportance.High, Resources.ZIPDEPLOY_Succeeded);
                         return true;
                     }
-                    else if (status == DeployStatus.Failed || status == DeployStatus.Unknown)
+                    else if (deploymentResponse?.Status == DeployStatus.Failed || deploymentResponse?.Status == DeployStatus.Unknown)
                     {
-                        Log.LogError(String.Format(Resources.ZIPDEPLOY_Failed, zipDeployPublishUrl, status));
+                        Log.LogError(string.Format(Resources.ZIPDEPLOY_Failed, zipDeployPublishUrl, deploymentResponse?.Status, deploymentResponse?.GetLogUrlWithId()));
```
Comment (vijayrkn) [inline src/WebSdk/Publish/Tasks/Tasks/ZipDeploy/ZipDeploy.cs:127]: What if the log Url is null? Can we handle the strings accordingly?

File: src/Tests/Microsoft.NET.Sdk.Publish.Tasks.Tests/Tasks/ZipDeploy/ZipDeploymentStatusTests.cs [hunk 2]
```diff
@@ -93,7 +99,26 @@ public async Task PollDeploymentStatusTest_ForValidResponses(HttpStatusCode resp
...
+            verifyStep(client, expectedDeployStatus == actualdeployStatus.Status);
+        }
+
+        [Theory]
+        [InlineData("https://mywebapp.scm.azurewebsites.net/api/deployments/latest/log", "id_1", "https://mywebapp.scm.azurewebsites.net/api/deployments/id_1/log")]
+        [InlineData("https://mywebapp.scm.azurewebsites.net/api/deployments/latest/log", "", "https://mywebapp.scm.azurewebsites.net/api/deployments/latest/log")]
+        [InlineData("https://mywebapp.scm.azurewebsites.net/api/deployments/latest", "id_2", "https://mywebapp.scm.azurewebsites.net/api/deployments/id_2")]
+        [InlineData("https://mywebapp.scm.azurewebsites.net/api/deployments/latest/diagnostics/log", "11223344", "https://mywebapp.scm.azurewebsites.net/api/deployments/11223344/diagnostics/log")]
+        [InlineData("https://latest.scm.azurewebsites.net/api/deployments/latest/log", "11223344", "https://latest.scm.azurewebsites.net/api/deployments/11223344/log")]
+        [InlineData("", "id_2", "")]
```
Comment (vijayrkn) [inline src/Tests/Microsoft.NET.Sdk.Publish.Tasks.Tests/Tasks/ZipDeploy/ZipDeploymentStatusTests.cs:111]: tests for null?

File: src/WebSdk/Publish/Tasks/Tasks/ZipDeploy/DeploymentResponse.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    public static class DeploymentResponseExtensions
+    {
+        public static string GetLogUrlWithId(this DeploymentResponse deploymentResponse)
+        {
+            if (deploymentResponse is null
+                || string.IsNullOrEmpty(deploymentResponse.LogUrl)
+                || string.IsNullOrEmpty(deploymentResponse.Id))
+            {
+                return deploymentResponse.LogUrl;
```
[the remark this answers is not in the document]
Comment (anvillan) [reply src/WebSdk/Publish/Tasks/Tasks/ZipDeploy/DeploymentResponse.cs:30]: No. We want to return whatever `LogUrl` is assigned to.

This covers the case for a null `Id`: as we cannot replace it for `latest`, return the `LogUrl` with no modifications.

[the remark this answers is not in the document]
Comment (anvillan) [reply src/WebSdk/Publish/Tasks/Tasks/ZipDeploy/DeploymentResponse.cs:30]: I will add the null check though.

**Review by vijayrkn — APPROVED** [review]

**Pull request merged**